tests/cortex_cli_test.py

```diff
@@ -336,40 +370,41 @@ def test_auth_logout_handles_keep_tokens_and_pid(config_dict, credentials):
             '--config-file', 'config.json',
             '--username', credentials['username'],
             '--password', credentials['password'],
-            '--no-daemon', # do not daemonize
+            '--no-refresh', # do not start token manager
             ])
         with open('tokens.json', 'r', encoding='utf-8') as file:
             tokens = json.loads(file.read())
 
-        # emulate having a daemon by adding a PID to tokens.json
-        tokens['pid'] = good_pid
-        with open('tokens.json', 'w', encoding='utf-8') as file:
-            file.write(json.dumps(tokens))
-
-        # user runs logout, but does not confirm
-        result = runner.invoke(cortex_cli,
-            ['auth', 'logout',
-            '--config-file', 'config.json',
-            '--keep-tokens',
-            ],
-            input = 'n')
-        assert result.exit_code == 0
-        assert 'Logout aborted.' in result.output
-
-        # user runs logout, confirms via --force
-        result = runner.invoke(cortex_cli,
-            ['auth', 'logout',
-            '--config-file', 'config.json',
-            '--keep-tokens',
-            '--force'
-            ])
-        assert result.exit_code == 0
-
-        # tokens file kept unchanged
-        with open('tokens.json', 'r', encoding='utf-8') as file:
-            tokens = json.loads(file.read())
-        assert tokens['access_token'] == expected_tokens['access_token']
-        assert tokens['refresh_token'] == expected_tokens['refresh_token']
+        # emulate a running daemon by spawning a Python subprocess and storing its PID
+        with subprocess.Popen('python') as p:
+            tokens['pid'] = p.pid
+            with open('tokens.json', 'w', encoding='utf-8') as file:
+                file.write(json.dumps(tokens))
+
+            # user runs logout, but does not confirm
+            result = runner.invoke(cortex_cli,
+                ['auth', 'logout',
+                '--config-file', 'config.json',
+                '--keep-tokens',
+                ],
+                input = 'n')
+            assert result.exit_code == 0
+            assert 'Logout aborted.' in result.output
+
+            # user runs logout, confirms via --force
+            result = runner.invoke(cortex_cli,
+                ['auth', 'logout',
+                '--config-file', 'config.json',
+                '--keep-tokens',
+                '--force'
+                ])
+            assert result.exit_code == 0
```

Comment:
Can we also check ``p`` was killed at this point?

Reply:
Good idea! But I didn't manage to find a robust, cross-platform way to start a process that's consistently killable by `psutil.Process.terminate()`. In the end I decided to mock that method instead, so now there is no real process killing going on. However, we are still using a real PID (== PID of the currently running Python-pytest process via `os.getpid()`) for the `psutil.pid_exists()` call.

As discussed earlier, we should write some integration or e2e tests against a real running daemon, but this goes beyond the scope of unit tests in Cortex CLI repository.